commit a299143ecca665246d139c8b2367f1f8ebd2d23a (refs/changes/64/3164/2)
Author: Jerry Jelinek <jerry.jelinek@joyent.com>
Date:   2018-01-09T17:29:13+00:00 (1 year, 9 months ago)
    
    OS-6553 CPU counter support doesn't work on many modern CPUs

diff --git a/manifest b/manifest
index b09a176577..b902813099 100644
--- a/manifest
+++ b/manifest
@@ -19,7 +19,7 @@
 # CDDL HEADER END
 #
 #
-# Copyright 2017 Joyent, Inc.
+# Copyright 2018 Joyent, Inc.
 # Copyright 2017 ASS-Einrichtungssysteme GmbH
 #
 # This lists all the files that illumos-joyent delivers as part of a build.
@@ -1858,11 +1858,22 @@ s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.29=pcbe.GenuineIntel.6.15
 s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.30=pcbe.GenuineIntel.6.15
 s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.31=pcbe.GenuineIntel.6.15
 s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.37=pcbe.GenuineIntel.6.15
+s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.42=pcbe.GenuineIntel.6.15
 s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.44=pcbe.GenuineIntel.6.15
 s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.45=pcbe.GenuineIntel.6.15
 s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.46=pcbe.GenuineIntel.6.15
 s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.47=pcbe.GenuineIntel.6.15
 s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.58=pcbe.GenuineIntel.6.15
+s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.60=pcbe.GenuineIntel.6.15
+s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.61=pcbe.GenuineIntel.6.15
+s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.62=pcbe.GenuineIntel.6.15
+s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.63=pcbe.GenuineIntel.6.15
+s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.69=pcbe.GenuineIntel.6.15
+s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.70=pcbe.GenuineIntel.6.15
+s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.71=pcbe.GenuineIntel.6.15
+s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.78=pcbe.GenuineIntel.6.15
+s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.79=pcbe.GenuineIntel.6.15
+s platform/i86pc/kernel/pcbe/amd64/pcbe.GenuineIntel.6.85=pcbe.GenuineIntel.6.15
 d platform/i86pc/kernel/sys 0755 root sys
 d platform/i86pc/kernel/sys/amd64 0755 root sys
 f platform/i86pc/kernel/sys/amd64/cpc 0755 root sys
diff --git a/usr/src/uts/intel/core_pcbe/Makefile b/usr/src/uts/intel/core_pcbe/Makefile
index 7f2d1ad8e5..1673cfe252 100644
--- a/usr/src/uts/intel/core_pcbe/Makefile
+++ b/usr/src/uts/intel/core_pcbe/Makefile
@@ -21,6 +21,7 @@
 #
 # Copyright (c) 2008, 2010, Oracle and/or its affiliates. All rights reserved.
 #
+# Copyright 2018, Joyent, Inc.
 #
 # This Makefile builds
 # the Intel Core Architecture Performance Counter BackEnd (PCBE).
@@ -42,9 +43,22 @@ SOFTLINKS	= pcbe.GenuineIntel.6.23 \
 			pcbe.GenuineIntel.6.30 \
 			pcbe.GenuineIntel.6.31 \
 			pcbe.GenuineIntel.6.37 \
+			pcbe.GenuineIntel.6.42 \
 			pcbe.GenuineIntel.6.44 \
+			pcbe.GenuineIntel.6.45 \
 			pcbe.GenuineIntel.6.46 \
-			pcbe.GenuineIntel.6.47
+			pcbe.GenuineIntel.6.47 \
+			pcbe.GenuineIntel.6.58 \
+			pcbe.GenuineIntel.6.60 \
+			pcbe.GenuineIntel.6.61 \
+			pcbe.GenuineIntel.6.62 \
+			pcbe.GenuineIntel.6.63 \
+			pcbe.GenuineIntel.6.69 \
+			pcbe.GenuineIntel.6.70 \
+			pcbe.GenuineIntel.6.71 \
+			pcbe.GenuineIntel.6.78 \
+			pcbe.GenuineIntel.6.79 \
+			pcbe.GenuineIntel.6.85
 ROOTSOFTLINKS	= $(SOFTLINKS:%=$(ROOT_PSM_PCBE_DIR)/%)
 
 #
diff --git a/usr/src/uts/intel/pcbe/core_pcbe.c b/usr/src/uts/intel/pcbe/core_pcbe.c
index bc571c45e9..b02b6fb2db 100644
--- a/usr/src/uts/intel/pcbe/core_pcbe.c
+++ b/usr/src/uts/intel/pcbe/core_pcbe.c
@@ -20,6 +20,7 @@
  */
 /*
  * Copyright (c) 2008, 2010, Oracle and/or its affiliates. All rights reserved.
+ * Copyright 2018 Joyent, Inc.
  */
 
 /*
@@ -1326,9 +1327,26 @@ core_pcbe_init(void)
 			/* Westmere */
 			case 37:
 			case 44:
+			/* Sandy Bridge */
+			case 42:
+			case 45:
 			/* Nehalem-EX */
 			case 46:
 			case 47:
+			/* Ivy Bridge */
+			case 58:
+			case 62:
+			/* Haswell */
+			case 60:
+			case 63:
+			case 69:
+			case 70:
+			/* Broadwell */
+			case 61:
+			case 71:
+			/* Skylake */
+			case 78:
+			case 85:
 				events_table = events_fam6_nhm;
 				break;
 			case 28:
